imap i/o timeout

Hi

I was curious about this functionality so I tested it out on a Synology NAS with Docker support.
By looking at the logs it seems it is working.

```
{"level":"info","msg":"Email received for info@account.netflix.com","time":"2026-07-13T16:20:17Z","trace_id":"fb323213-be62-4c6f-8c3c-d829d695f10f"} 
{"level":"info","msg":"Found 1 unseen email(s) to process","time":"2026-07-13T16:20:17Z"} 
```
```
{"level":"info","msg":"IMAP connection established successfully","time":"2026-07-12T14:33:02Z"} 
{"level":"info","msg":"Starting Netflix email verification process (IMAP IDLE mode)","time":"2026-07-12T14:33:02Z"} 
```

But when a targeted email to process comes in it stops/times out somewhere
```
/app/cmd/main.go:57 +0x1de 
main.main() 
/app/cmd/main.go:133 +0x232 
main.fetchAndProcessEmails(0x225f1874c0a0, 0x225f188201e0, 0x225f18800360) 
/app/internal/emailprocessor/processor.go:67 +0x193 
netflix-household-validator/internal/emailprocessor.(*Processor).ProcessEmail(0x225f18ac9e08, 0x17) 
/app/internal/netflix/service.go:56 +0x2b2 
netflix-household-validator/internal/netflix.(*Service).HandleEmail(0x225f188201e0, 0x225f18974000) 
/app/internal/netflix/browser_rod.go:37 +0x1ff 
netflix-household-validator/internal/netflix.(*RodBrowser).OpenUpdatePrimaryLocation(0xfc4860, {0x225f18a4038b, 0x1d7}, {0x225f18844840, 0x24}) 
/app/internal/netflix/browser_rod.go:115 +0x671 
netflix-household-validator/internal/netflix.(*RodBrowser).attemptOpenLink(0x225f18854000?, {0x225f18a4038b, 0x1d7}, 0x1, {0x225f18844840, 0x24}) 
/go/pkg/mod/github.com/go-rod/rod@v0.116.2/must.go:71 +0xc9 
github.com/go-rod/rod.(*Browser).MustPage(0x225f188b2480, {0x225f188ddb30?, 0x7?, 0x225f1892ef50?}) 
/go/pkg/mod/github.com/go-rod/rod@v0.116.2/must.go:36 +0x5c 
netflix-household-validator/internal/netflix.(*RodBrowser).attemptOpenLink.New.(*Browser).WithPanic.genE.func4({0x225f18ad4480?, 0x1d7?, 0x0?}) 
/go/pkg/mod/github.com/go-rod/rod@v0.116.2/lib/utils/utils.go:68 +0x1d 
github.com/go-rod/rod/lib/utils.init.func2({0x885860?, 0x225f18ad4400?}) 
goroutine 1 [running]: 
panic: navigation failed: net::ERR_ABORTED 
imap/client: 2026/07/13 16:20:47 error reading response: read tcp 192.168.158.10:38946->188.68.47.240:993: i/o timeout 
{"level":"info","msg":"Attempt 1/3 (fresh browser \u0026 profile)","time":"2026-07-13T16:20:17Z","trace_id":"fb323213-be62-4c6f-8c3c-d829d695f10f"} 
{"level":"info","msg":"Open page with rod: https://www.netflix.com/account/update-primary-location?g=%2A%2A%2A%2A%2A%2A\u0026lkid=UPDATE_HOUSEHOLD_REQUESTED_OTP_CTA\u0026lnktrk=EVO\u0026nftoken=%2A%2A%2A%2A%2A%2A\u0026operation=update","time":"2026-07-13T16:20:17Z","trace_id":"fb323213-be62-4c6f-8c3c-d829d695f10f"} 
```

Do you know why?
How can I debug/investigate?

The URL in the mail looks actually different:
https://www.netflix.com/account/update-primary-location?nftoken=LONGTOKEN=EVO&operation=update&lkid=UPDATE_HOUSEHOLD_REQUESTED_OTP_CTA

thanks

--- a/internal/netflix/browser_rod.go
+++ b/internal/netflix/browser_rod.go
@@ -1,6 +1,7 @@
 package netflix
 
 import (
+	"fmt"
 	"net/url"
 	"netflix-household-validator/internal/models"
 	"os"
@@ -71,11 +72,21 @@ func (rb *RodBrowser) attemptOpenLink(
 	link string,
 	attempt int,
 	traceID string,
-) (models.BrowserResult, error) {
+) (result models.BrowserResult, err error) {
 	activeRodSessions.Add(1)
 	defer activeRodSessions.Add(-1)
 
 	locallog := logging.Log.WithField("trace_id", traceID)
+
+	// Recover from any panic in the browser automation so a single bad link
+	// can never crash the whole validator / IMAP loop.
+	defer func() {
+		if r := recover(); r != nil {
+			locallog.Errorf("Recovered from panic in attemptOpenLink: %v", r)
+			result = models.ResultFailed
+			err = fmt.Errorf("panic in browser automation: %v", r)
+		}
+	}()
 
 	tmpDir, err := os.MkdirTemp("", "rod-netflix-*")
 	if err != nil {
@@ -112,15 +123,23 @@ func (rb *RodBrowser) attemptOpenLink(
 		return models.ResultFailed, err
 	}
 
-	page := browser.MustPage(link)
+	page, err := browser.Page(proto.TargetCreateTarget{URL: link})
+	if err != nil {
+		locallog.WithError(err).Error("failed to open page")
+		return models.ResultFailed, err
+	}
 	defer func() { _ = page.Close() }()
 
-	page.MustWaitLoad()
+	if err := page.WaitLoad(); err != nil {
+		locallog.WithError(err).Warnf("Attempt %d: wait load failed (navigation may have been redirected)", attempt)
+	}
 
 	// Try to accept cookie banner if present
 	if cookieBtn, err := page.Timeout(5 * time.Second).Element("#onetrust-accept-btn-handler"); err == nil {
 		locallog.Info("Cookie banner detected, accepting")
-		cookieBtn.MustClick()
+		if clickErr := cookieBtn.Click(proto.InputMouseButtonLeft, 1); clickErr != nil {
+			locallog.WithError(clickErr).Warn("Failed to click cookie banner")
+		}
 	}
 
 	outcome, err := racePageElements(page, 15*time.Second)

